libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+pub enum PostHogFlagFilterPropertyValue {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+    List(Vec<String>),
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariate {
+    variants: Vec<PostHogLocalEvaluationFlagMultivariateVariant>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariateVariant {
+    key: String,
+    rollout_percentage: f64,
+}
+
+pub struct PostHogClient {
+    server_api_key: String,
+    client_api_key: String,
+    project_id: String,
+    private_api_url: String,
+    public_api_url: String,
+    client: Arc<reqwest::Client>,
+}
+
+pub struct FeatureStore {
+    flags: HashMap<String, PostHogLocalEvaluationFlag>,
+}
+
+impl Default for FeatureStore {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl FeatureStore {
+    pub fn new() -> Self {
+        Self {
+            flags: HashMap::new(),
+        }
+    }
+
+    pub fn update_flags(&mut self, flags: Vec<PostHogLocalEvaluationFlag>) {
+        self.flags.clear();
+        for flag in flags {
+            if flag.active {
+                self.flags.insert(flag.key.clone(), flag);
+            }
+        }
+    }
+
+    /// Generate a consistent hash for a user ID (e.g., tenant ID).
+    ///
+    /// The implementation is different from PostHog SDK. In PostHog SDK, it is sha1 of `user_id.distinct_id.salt`.
+    /// However, as we do not upload all of our tenant IDs to PostHog, we do not have the PostHog distinct_id for a
+    /// tenant.
+    fn consistent_hash(user_id: &str) -> f64 {
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(user_id);
+        let hash = hasher.finalize();
+        let hash_int = u64::from_le_bytes(hash[..8].try_into().unwrap());
+        hash_int as f64 / u64::MAX as f64
+    }
+
+    /// Evaluate a condition. Returns `None` if the condition cannot be evaluated due to parsing error or missing
+    /// property.
+    fn evaluate_condition(
+        &self,
+        operator: &str,
+        provided: &PostHogFlagFilterPropertyValue,
+        requested: &PostHogFlagFilterPropertyValue,
+    ) -> Option<bool> {
+        match operator {
+            "exact" => {
+                let PostHogFlagFilterPropertyValue::String(provided) = provided else {
+                    // Left should be a string
+                    return None;
+                };
+                let PostHogFlagFilterPropertyValue::List(requested) = requested else {
+                    // Right should be a list of string
+                    return None;
+                };
+                Some(requested.contains(provided))
+            }
+            "lt" | "gt" => {
+                let PostHogFlagFilterPropertyValue::String(requested) = requested else {
+                    // Right should be a string
+                    return None;
+                };
+                let Ok(requested) = requested.parse::<f64>() else {
+                    return None;
```

Comment:
Is it ok that we return `None` and not real errors here and there?
Probably we want to log if there is some error in evaluation because usually it means bugs

Reply:
Yep I'll return real errors there